src/nbl/video/CVulkanConnection.h

```diff
@@ -3,43 +3,83 @@
 
 #include <volk.h>
 #include "nbl/video/IAPIConnection.h"
-#include "nbl/video/surface/CSurfaceVKWin32.h"
 #include "nbl/video/CVulkanPhysicalDevice.h"
 
-namespace nbl {
-namespace video
+#if defined(_NBL_PLATFORM_WINDOWS_)
+#   include "nbl/ui/IWindowWin32.h"
+#   include "nbl/video/surface/CSurfaceVKWin32.h"
+#endif
+
+namespace nbl::video
 {
 
 class CVulkanConnection final : public IAPIConnection
 {
 public:
-    CVulkanConnection(uint32_t appVer, const char* appName)
+    CVulkanConnection(core::smart_refctd_ptr<system::ISystem>&& sys, uint32_t appVer,
+        const char* appName, const SDebugCallback& dbgCb) : IAPIConnection(std::move(sys))
     {
         VkResult result = volkInitialize();
         assert(result == VK_SUCCESS);
+        
+        VkApplicationInfo applicationInfo = { VK_STRUCTURE_TYPE_APPLICATION_INFO };
+        applicationInfo.apiVersion = VK_MAKE_VERSION(1, 1, 0);
+        applicationInfo.engineVersion = NABLA_VERSION_INTEGER;
+        applicationInfo.pEngineName = "Nabla";
+        applicationInfo.applicationVersion = appVer;
+        applicationInfo.pApplicationName = appName;
+        
+        // Todo(achal): Get this from the user
+        const std::vector<const char*> requiredInstanceLayerNames({ "VK_LAYER_KHRONOS_validation" });
+        assert(areAllInstanceLayersAvailable(requiredInstanceLayerNames));
```

Comment:
this is why you essentially want to hoist entire constructors to a `create` method, at least there you have the option to bail and return `nullptr`